sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,6 +119,39 @@ object IntervalUtils {
     }
   }
 
+  private def throwIllegalIntervalFormatException(
+      input: UTF8String,
+      startFiled: Byte,
+      endField: Byte,
+      intervalStr: String,
+      typeName: String,
+      fallBackNotice: Option[String] = None) = {
+    throw new IllegalArgumentException(
+      s"Interval string does not match $intervalStr format of " +
+        s"${supportedFormat((startFiled, endField))
+          .map(format => s"`$format`").mkString(", ")} " +
+        s"when cast to $typeName: ${input.toString}" +
+        s"${fallBackNotice.map(s => s", $s").getOrElse("")}")
+  }
+
+  private def checkStringIntervalType(
+      input: UTF8String,
+      startField: Byte,
+      endField: Byte,
+      targetInterval: DataType): Unit = {
+    val (intervalStr, typeName, targetStartField, targetEndField) = targetInterval match {
+      case DT(startField, endField) =>
+        ("day-time", targetInterval.typeName, startField, endField)
+      case YM(startField, endField) =>
+        ("year-month", targetInterval.typeName, startField, endField)
+    }
+    if (startField != targetStartField || endField != targetEndField) {
```

Comment:
Do you catch the case when: `startField == targetStartField && endField == targetEndField` but `startField > endField`?

Reply:
In current code  if `targetStartField == inputStartField && targetEndField == inputEndField`
Since target code from cast's target DT/YM type, it should be valid. so it's necessary to check this here?